rubix-core/src/main/java/com/qubole/rubix/core/NonLocalReadRequestChain.java

```diff
@@ -193,4 +197,45 @@ private int directReadRequest(int index)
     directReadChain = null;
     return (totalRead + directRead);
   }
+
+  @Override
+  public void updateCacheStatus(String remotePath, long fileSize, long lastModified, int blockSize, Configuration conf)
+  {
+    if (CacheConfig.isDummyModeEnabled(conf)) {
+      RetryingBookkeeperClient bookKeeperClient = null;
+      try {
+        bookKeeperClient = new BookKeeperFactory().createBookKeeperClient(remoteNodeName, conf);
+        for (ReadRequest readRequest : readRequests) {
+          long startBlock = toBlock(readRequest.getBackendReadStart());
+          long endBlock = toBlock(readRequest.getBackendReadEnd() - 1) + 1;
+          log.debug("Dummy Mode: Non-local Read Request: Setting cached from : " + startBlock + " block to : " + endBlock);
+          CacheStatusRequest request = new CacheStatusRequest(remotePath, fileSize, lastModified, startBlock, endBlock, clusterType);
+          bookKeeperClient.getCacheStatus(request);
```

Comment:
Add a comment why getCacheStatus is needed

Reply:
Okay